include/sys/trace.h

```diff
@@ -43,19 +43,23 @@
  */
 #undef DTRACE_PROBE1
 #define	DTRACE_PROBE1(name, t1, arg1) \
-	trace_zfs_##name((arg1))
+	if (arg1) \
+		trace_zfs_##name((arg1))
 
 #undef DTRACE_PROBE2
 #define	DTRACE_PROBE2(name, t1, arg1, t2, arg2) \
-	trace_zfs_##name((arg1), (arg2))
+	if (arg1 && arg2) \
```

Comment:
@behlendorf @loli10K uploaded the pull request for the patch suggested by @loli10K ...The only additional change is the size of the buffer is increased to 512 bytes. The reason for doing it that the spa name itself can be up to 256 bytes. We ourselves are using bigger names for spa. Sample o/p:

 dbuf_evict-4115  [005] ....   460.018195: zfs_dbuf__evict__one: dbuf { spa "zpool-XOXO-WT-e1c0d782-ca7f-4916-bf68-8fc0650a1172-fd66d34a-43c1-44de-8970-5e659863b235" objset 91 object 4381207 level 0 blkid 0 offset 0 size 50688 state 4 holds 0 }
dbuf_evict-4115  [003] ....  1394.738505: zfs_dbuf__evict__one: dbuf { NULL }